### PR TITLE
[Syntax] Parse 'switch' statement

### PR DESCRIPTION
**Pull request by rintaro**
Resolves #49466
CC: @nkcsgexi @harlanhaskins

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: utils/gyb_syntax_support/DeclNodes.py [hunk 1]
```diff
@@ -96,11 +96,13 @@
...
     Node('IfConfigDecl', kind='Decl',
-         traits=['WithStatements'],
          children=[
              Child('PoundIf', kind='PoundIfToken'),
              Child('Condition', kind='Expr'),
-             Child('Statements', kind='CodeBlockItemList'),
+             Child('Elements', kind='Syntax',
+                   node_choices=[
+                      Child('Statements', kind='CodeBlockItemList'),
+                      Child('SwitchCases', kind='SwitchCaseList')]),
```
Comment (harlanhaskins) [inline utils/gyb_syntax_support/DeclNodes.py]: I wonder if there’s anything we could do to try to merge these

**Review by harlanhaskins — APPROVED** [review]
LGTM!

File: utils/gyb_syntax_support/DeclNodes.py (hunk 1 above)
Comment (rintaro) [inline utils/gyb_syntax_support/DeclNodes.py]: Dropping `WithStatements` trait from `IfConfigDecl`.
But separating `IfConfigDecl` only for switch-cases (i.e. `SwitchCaseIfConfig`) doesn't seem the right choice.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by harlanhaskins** [comment]
Can you also update `lib/Syntax/Status.md`?

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by nkcsgexi — APPROVED** [review]
LGTM, thank you for clearing the statement section!

**Comment by jrose-apple** [comment]
Ah, please make sure I remember to update this after #14382 goes in!

**Pull request merged**